src/flint/fmpz.jl

```diff
@@ -85,6 +85,44 @@ ifelse(stop >= start, stop, start - one(ZZRingElem))
 
 Base.:(:)(a::ZZRingElem, b::ZZRingElem) = ZZRingElemUnitRange(a, b)
 
+struct ZZOneTo <: AbstractUnitRange{ZZRingElem}
+  stop::ZZRingElem
+end
+
+Base.OneTo(n::ZZRingElem) = ZZOneTo(n)
+
+Base.eltype(::Type{ZZOneTo}) = ZZRingElem
+Base.first(::ZZOneTo) = ZZ(1)
+Base.last(r::ZZOneTo) = r.stop
+
+function Base.length(r::ZZOneTo)
+  s = r.stop
+  if fits(Int, s)
+      return Int(s)
+  else
+      return BigInt(s)
+  end
+end
+
+function Base.getindex(r::ZZOneTo, i::Integer)
+  if i < 1 || i > length(r)
+      throw(BoundsError(r, i))
+  end
```

Comment:
Bounds checks should be annotated by `@boundscheck` (read up in the documentation as to what that does). Also, `length(r)` implies a conversion; better to directly compare `i` to `r.step`, then we can potentially generate better code
```suggestion
  @boundscheck  1 <= i <= r.stop || Base.throw_boundserror(r, i)
```